### PR TITLE
page component

### PR DESCRIPTION
**Pull request by rosscreighton**
TL;DR: don't be confused by the branch name.

the name of this branch is confusing. i called it `application-layout` because i was originally thinking of this page component as a 'layout' because it renders a header and footer like a layout would. however i was thinking about it and this component will do more than provide standardized layout - it will handle user data, analytics, etc., so it is a component. 

alos, right now, just a `Page` component will serve us well, but in the future as we add more functionality to this component we may want to break it out into an `Application` component and `Page` component.

there are also some performance tweaks to `AlbumsContainer` in this PR, which cut the `AlbumPage` re-renders from 4 to 2. see commit messages for more detail.

### CODE REVIEW

File: imports/ui/containers/AlbumsContainer.jsx [hunk 1]
```diff
@@ -23,8 +23,8 @@ export default AlbumsContainer = createContainer(({ params }) => {
   }
 
   return {
-    album: albumReady ? album : {},
```
Comment (sameckmeier) [inline imports/ui/containers/AlbumsContainer.jsx]: v nice

File: imports/ui/containers/AlbumsContainer.jsx [hunk 2]
```diff
@@ -10,9 +10,9 @@ const spotifyAlbum = new ReactiveVar(null);
 export default AlbumsContainer = createContainer(({ params }) => {
   const albumsHandle = Meteor.subscribe('album', params.slug);
   const album = Albums.findOne({slug: params.slug});
+  const spotifyAlbumLoading = spotifyAlbum.get() === null;
```
Comment (sameckmeier) [inline imports/ui/containers/AlbumsContainer.jsx]: right right, nice

File: imports/ui/containers/AlbumsContainer.jsx [hunk 3]
```diff
@@ -9,10 +9,12 @@ const spotifyAlbum = new ReactiveVar(null);
 
 export default AlbumsContainer = createContainer(({ params }) => {
   const albumsHandle = Meteor.subscribe('album', params.slug);
-  const album = Albums.findOne({slug: params.slug});
+  const albumReady = albumsHandle.ready();
```
Comment (sameckmeier) [inline imports/ui/containers/AlbumsContainer.jsx]: fantastic change. we should probs remove the second, embedded albumChange

File: imports/ui/pages/AlbumsPage.jsx [hunk 4]
```diff
@@ -2,9 +2,12 @@ import React, { Component, PropTypes } from 'react';
 
 export default class AlbumsPage extends Component {
   render() {
+    const { album, spotifyAlbum } = this.props;
```
Comment (sameckmeier) [inline imports/ui/pages/AlbumsPage.jsx]: i really like this. we should make a card with a checklist of some primary code quality todos, one of which is unpacking props/args
